src/components/select/Select.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable react/sort-comp */
+import React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import SelectHeader from "./SelectHeader";
+import SelectList from "./SelectList";
+
+/**
+ * @class Select - It is the wrapper for the Select Header and the Select List
+ */
+class Select extends React.PureComponent {
+  /**
+   * @property {array} options - List of options
+   * @property {string} placeholder - Placeholder for the select
+   * @property {string} icon - Name of the icon if the placeholder needs one
+   * @property {string} className - ClassName if you need to customize
+   */
+  constructor(props) {
+    super(props);
+    this.selectRef = React.createRef();
+    const { placeholder, icon } = this.props;
```

Comment:
Manten las asignaciones de this juntas, es decir para esta linea arriba de la de arriba.